sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/implementation/AzureConfigurableImpl.java

```diff
@@ -125,7 +126,15 @@ public T withConfiguration(Configuration configuration) {
         return (T) this;
     }
 
+    public void withHttpPipeline(HttpPipeline httpPipeline) {
+        Objects.requireNonNull(httpPipeline);
+        this.httpPipeline = httpPipeline;
+    }
+
     protected HttpPipeline buildHttpPipeline(TokenCredential credential, AzureProfile profile) {
+        if (this.httpPipeline != null) {
+            return httpPipeline;
+        }
         Objects.requireNonNull(credential);
```

Comment:
Could use some comment here to describe the necessity of the (a bit strange) code.